joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/datastructures/CSharpScope.scala

```diff
@@ -47,7 +48,16 @@ class CSharpScope(typeMap: TypeMap) extends Scope[String, DeclarationNew, ScopeT
     .exists(x => x.scopeNode.isInstanceOf[MethodScope] || x.scopeNode.isInstanceOf[TypeLikeScope])
 
   def tryResolveTypeReference(typeName: String): Option[String] = {
-    typesInScope.find(_.name.endsWith(typeName)).flatMap(typeMap.namespaceFor).map(n => s"$n.$typeName")
+    typesInScope
+      .find(_.name.endsWith(typeName))
+      .flatMap(typeMap.namespaceFor)
+      .map(n => {
+        boundary:
```

Comment:
I'm not a fan of `boundary-break`, this is a side-effect which we would like to avoid. Why not simply continue mapping?

Reply:
Also, you can use `==` in Scala, strings are equated by value

Reply:
The IDE threw a warning, thought we should avoid using if this is discontinued. Will revert it back if you think it's not a good idea.